app/src/main/java/com/github/geohunt/app/model/LiveLazyRef.kt

```diff
@@ -0,0 +1,19 @@
+package com.github.geohunt.app.model
+
+
+abstract class LiveLazyRef<T>: BaseLazyRef<T>() {
+    /**
+     * Starts listening the database for changes.
+     */
+    abstract fun addListener(callback: (T) -> Any?): LiveLazyRefListener
+}
+
+/**
+ * Represents a listener.
+ */
+abstract class LiveLazyRefListener {
+    /**
+     * Stops listening.
+     */
+    abstract fun stop()
+}
```

Comment:
> The idea of having live state in LazyRef was a bit the reason why I created this class in the first place.

This was not written anywhere so I could not know :shrug:. I thought the main reason was to actually hold a "reference" to a database element (as you mentionned).

> Ultimately if we have LazyRef everywhere is because we want to have a system of "updates" and "listeners". On that note my own branch would beneficiate such a system.

Then, every lazy ref should implement live lazy refs? This is not a "good" idea. I wish I could use mixins…

> On that note my own branch would beneficiate such a system.

OK.

> Because normally no one has used LazyRef directly within a UI context you should only have to update (along with their instantiation) the RememberLazyRef API.

No, adding a disposable effect to every component that remembers a lazy ref is not a good idea: creating too many listeners for every components is too much. It should be used only when necessary.

Reply:
Thinking back to what you said, I do not think that every lazy ref should be live. For instance, a challenge's image.

Reply:
And again, having a Live lazy ref requires a specific implementation (that could be too much for all the existing lazy refs).

Reply:
Update?